docs/DeveloperGuide.md

```diff
@@ -1,34 +1,422 @@
+
+2. James receive the following feedback:
+
+   ```
+   root :
+   addc misc
+   SUCCESS!! Category misc is created.
+   ```
+
+
+
+Below is a *sequence diagram* to illustrate the above example scenario.  
+
+```
+to-do: add the sequence diagram
+```
+
+
+
+### 2. Delete Feature  
+#### Overview  
+The **delete** feature deletes *modules*, *categories* and *tasks* from the Module, Category and Task List respectively.   
+When the user first requests to execute the **delete** command to delete a directory by providing its name, the application will first filter for directories with matching names. From here, there are **three** possible outcomes:  
+
+1. There are **no** matches --  Nothing is deleted.
+2. There is **one** match -- A prompt will be given to the user to confirm the deletion.
+3. There are **multiple** matches -- The list of matches will be shown to the user, and the user chooses which ones to delete. A further prompt will be given to confirm the deletion(s).
+
+#### Feature Implementation  
+![Delete Command Class Diagram](images/Delete_Command_Class_Diagram.png)
+<span style="color: green"><small><i>Figure <b>Delete Command Class Diagram</b></i></small></span>
+
+<br>
+
+The `DeleteModuleCommand`, `DeleteCategoryCommand` and `DeleteTaskCommand` classes in the application facilitates this **delete** feature. They are in charge of deleting *modules*, *categories* and *tasks* respectively. <br>  
+
+As shown in the figure above, those three classes each extends from the abstract `DeleteCommand` class.  They also [override](#) the `DeleteCommand` class's `executeInitialDelete()` method, which role is to prepare the necessary messages or prompts for the user depending on the number of filtered matches. <br>  
+
+The `DeleteCommand` class in turn extends the `FilterCommand` abstract class. The `FilterCommandClass` contains the following vital methods for filtering:  
+- `createFilteredModuleList()` -- Creates an `ArrayList` of the filtered *modules*.  
+- `createFilteredCategoryist()` -- Creates an `ArrayList` of the filtered *categories*.  
+- `createFilteredTaskList()` -- Creates an `ArrayList`of the filtered *tasks*.  
+
+Lastly, the `FilterCommand` class extends the abstract `Command` class that contains the `execute()` method to execute the actual **delete** command.  
+<br>  
+
+![Prompt Command Class Diagram](images/Prompt_Class_Diagram.png)
```

Comment:
YES this is perfect amount of information! Clear and minimalist but still provide enough information for the reader to understand the class diagram